src/generator.cpp

```diff
@@ -1222,18 +1247,42 @@ void Generator::GeneratorImpl::processEquationAst(const GeneratorEquationAstPtr
 
                 mGenerator->addError(err);
             } else {
-                mVoi = variable;
+                // We have found our variable of integration, but this may not
+                // be the one defined in our first component (i.e. the component
+                // under which we are likely to expect to see the variable of
+                // integration to be defined), so go through our components and
+                // look for the first occurrence of our variable of integration.
```

Comment:
What constitutes our first occurrence of the VOI? What metric do you use to determine the first occurrence?

Reply:
Directionality has been removed in CellML 2.0, so we can't tell where a variable was originally defined. However, when it comes to the VOI, it has historically been defined in the first component of a CellML document. So, we go through the different components and look for the first one that defines the VOI, and we consider that component as being the component in which the VOI was "originally" defined. So, yes, very much a heuristic approach.